ncm-sysconfig/src/main/perl/sysconfig.pm

```diff
@@ -87,7 +86,21 @@ sub Configure
             # Loop over the pairs adding the information to the file.
             for my $key (sort keys %$pairs) {
                 if ($key ne 'prologue' && $key ne 'epilogue') {
-                    $contents .= "$key=$pairs->{$key}\n";
+                    my $value = $pairs->{$key};
+                    # Remove any leading or trailing whitespace from the value
+                    $value =~ s/^\s+|\s+$//g;
+                    # Quote values containing whitespace.
+                    # Only if they have not already been quoted and do not look like an array.
+                    # If a value contains quotes, use a different quotation mark.
+                    # Values without whitespace should not be modified.
+                    if ($value !~ m/^".*"$/ && $value !~ m/^'.*'$/ && $value !~ m/^\(.*\)$/ && $value =~ m/\s/) {
```

Comment:
i'm not sure what tis does "for array", but you should only do this on scalars (and probabnly report an error for anything else:
```perl
my $ref = ref($value);
if ($ref) {
    error("something something should not pass non-scalar to sysconfig: got type $ref")
} else {
   # do magic wrt quotes and whitespace
}
```

Reply:
nvm, i was totally confused